fferr/data.go

```diff
@@ -52,18 +52,22 @@ type DatasetAlreadyExistsError struct {
 	baseError
 }
 
-func NewDataTypeNotFoundError(valueType string, err error) *DataTypeNotFoundError {
+func NewDataTypeNotFoundError(value any, err error) *DataTypeNotFoundError {
 	if err == nil {
 		err = fmt.Errorf("datatype not found")
 	}
 	baseError := newBaseError(err, DATATYPE_NOT_FOUND, codes.NotFound)
-	baseError.AddDetail("value_type", valueType)
+	baseError.AddDetail("value_and_type", fmt.Sprintf("%#v %T", value, value))
```

Comment:
if you wanted to, you could do baseError.AddDetails("value", fmt.Sprintf("%#v"), "type", "fmt.Sprintf("%T"))